plugin/src/main/java/io/jenkins/plugins/casc/SecretSourceResolver.java

```diff
@@ -55,17 +54,36 @@
      * @return Encoded string
      * @since 1.25
      */
-    public static String encode(@CheckForNull String toEncode) {
+    public String encode(@CheckForNull String toEncode) {
         if (toEncode == null) {
             return null;
         }
         return toEncode.replace(enclosedBy, escapeEnclosedBy);
     }
 
+    /**
+     * Resolve string with potential secrets
+     *
+     * @param context Configuration context
+     * @param toInterpolate potential variables that need to revealed
+     * @return original string with any secrets that could be resolved if secrets could not be resolved they will be defaulted to default value defined by ':-', otherwise default to empty String secrets are defined as anything enclosed by '${}'
+     * @deprecated use ${link {@link #resolve(String)}} instead.
+     * @since 1.42
+     */
+    @Deprecated
```

Comment:
Could potentially remove this as usually this is only called directly from test packages.

Depeneds if we are okay with yet another breakage 😆